### PR TITLE
Convert create org to dialog

### PR DESCRIPTION
**Pull request by aaronkchsu**

### CODE REVIEW

File: src/api/graphql.generated.tsx [hunk 1]
```diff
@@ -62,11 +62,6 @@ export interface AttachUserToSecurityGroupInput {
   request: AttachUserToSecurityGroup;
 }
 
-export interface BaseNodeConsistencyInput {
```
Comment (CarsonF) [inline src/api/graphql.generated.tsx:65]: Your server code is behind causing this generated schema to regress. Please pull the latest on API and re-gen this.

File: src/scenes/Organizations/Create/CreateOrganization.tsx [hunk 2]
```diff
@@ -10,23 +12,42 @@ import {
...
+  const [close, setClose] = useState<boolean>(false);
+  const navigate = useNavigate();
   const submit: Props['onSubmit'] = async (input) => {
     try {
+      setClose(true);
       await createOrg({
         variables: { input },
       });
+      const action = () => (
+        <Fragment>
```
Comment (CarsonF) [inline src/scenes/Organizations/Create/CreateOrganization.tsx:24]:
```suggestion
```

File: src/scenes/Organizations/Create/CreateOrganization.tsx [hunk 3]
```diff
@@ -10,23 +12,42 @@ import {
...
+          <Button
+            color="secondary"
+            variant="outlined"
+            onClick={() => {
+              // should be navigated to new org created.
+              navigate('/listOrgs', { replace: true });
+            }}
+          >
+            List
+          </Button>
```
Comment (CarsonF) [inline src/scenes/Organizations/Create/CreateOrganization.tsx:34]:
```suggestion
          <ButtonLink
            color="inherit"
            to={`/organizations/${org.id}`}
          >
            View
          </ButtonLink>
```

File: src/scenes/Home/index.tsx [hunk 4]
```diff
@@ -1,10 +1,26 @@
...
+    <div>
+      <div>Welcome, {session?.displayFirstName.value ?? 'Friend'}</div>
+      <Button onClick={() => setOpen(true)} color="primary" variant="contained">
+        CreateOrg
+      </Button>
+      <CreateOrganization
+        open={open}
+        onClose={(open: boolean) => {
+          setOpen(open);
+        }}
```
Comment (CarsonF) [inline src/scenes/Home/index.tsx:22]:
```suggestion
        onClose={() => {
          setOpen(false);
        }}
```

File: src/scenes/Organizations/Create/CreateOrganizationForm.tsx [hunk 5]
```diff
@@ -7,25 +13,44 @@ import { SubmitButton, SubmitError, TextField } from '../../../components/form';
...
+        <Form {...props}>
+          {({ handleSubmit }) => (
+            <form onSubmit={handleSubmit}>
+              <SubmitError />
+              <TextField
+                name="organization.name"
+                label="Name"
+                placeholder="Enter organization name"
+              />
+              <SubmitButton />
```
Comment (CarsonF) [inline src/scenes/Organizations/Create/CreateOrganizationForm.tsx:44]: `SubmitButton` should be next to the cancel button. You'll need to move the `Form` component higher so `SubmitButton` can work inside of `DialogActions`

File: src/scenes/Organizations/Create/CreateOrganizationForm.tsx [hunk 6]
```diff
@@ -7,25 +13,44 @@ import { SubmitButton, SubmitError, TextField } from '../../../components/form';
...
+            </form>
+          )}
+        </Form>
+      </DialogContent>
+      <DialogActions onClick={() => onClose(close)}>
```
Comment (CarsonF) [inline src/scenes/Organizations/Create/CreateOrganizationForm.tsx:49]: The onClick should be on the cancel button not here.

File: src/scenes/Organizations/Create/CreateOrganization.tsx [hunk 7]
```diff
@@ -10,23 +12,42 @@ import {
...
+        </Fragment>
+      );
       enqueueSnackbar(
         `Successfully created org with name: ${input.organization.name}`,
         {
           variant: 'success',
+          autoHideDuration: 2000,
```
Comment (CarsonF) [inline src/scenes/Organizations/Create/CreateOrganization.tsx:41]: The default duration is fine.
```suggestion
```

File: src/scenes/Organizations/Create/CreateOrganization.tsx [hunk 8]
```diff
@@ -10,23 +12,42 @@ import {
...
+              // should be navigated to new org created.
+              navigate('/listOrgs', { replace: true });
+            }}
+          >
+            List
+          </Button>
+        </Fragment>
+      );
       enqueueSnackbar(
         `Successfully created org with name: ${input.organization.name}`,
```
Comment (CarsonF) [inline src/scenes/Organizations/Create/CreateOrganization.tsx:38]:
```suggestion
        `Created organization: ${input.organization.name}`,
```

File: src/scenes/Organizations/Create/CreateOrganization.tsx [hunk 9]
```diff
@@ -10,23 +12,42 @@ import {
 export const CreateOrganization = (props: Except<Props, 'onSubmit'>) => {
   const [createOrg] = useCreateOrganizationMutation();
   const { enqueueSnackbar } = useSnackbar();
-
+  const [close, setClose] = useState<boolean>(false);
```
Comment (CarsonF) [inline src/scenes/Organizations/Create/CreateOrganization.tsx:15]: The generic can be inferred here.
```suggestion
  const [close, setClose] = useState(false);
```

File: src/scenes/Home/index.tsx (hunk 4 above)
Comment (aaronkchsu) [reply src/scenes/Home/index.tsx:22]: We should leave this because we can't close dialog during processing create org.

Comment (CarsonF) [reply src/scenes/Home/index.tsx:22]: That's true. But this change doesn't prevent that.

For that you need to set `disableBackdropClick` on `Dialog` to true will submitting. And also disable the cancel button.

Comment (aaronkchsu) [reply src/scenes/Home/index.tsx:22]: Actually, first one I did can prevent that. BTW your option looks better. I will change it

**Pull request merged**